tools/wptserve/wptserve/sslutils/openssl.py

```diff
@@ -114,16 +114,18 @@ def make_subject(common_name,
     return "".join(rv)
 
 def make_alt_names(hosts):
-    rv = []
-    for name in hosts:
-        rv.append("DNS:%s" % name)
-    return ",".join(rv)
+    return ",".join("DNS:%s" % host for host in hosts)
+
+def make_name_constraints(hosts):
+    return ",".join("permitted;DNS:%s" % host for host in hosts)
 
 def get_config(root_dir, hosts, duration=30):
     if hosts is None:
         san_line = ""
+        constraints_line = ""
     else:
         san_line = "subjectAltName = %s" % make_alt_names(hosts)
+        constraints_line = "nameConstraints = " + make_name_constraints(hosts)
```

Comment:
This will lead to trying to put `nameConstraints` on the server cert, rather than the CA cert.

From `_generate_ca` we call `get_config` with `hosts=None`:

```
  web-platform-tests/tools/wptserve/wptserve/sslutils/openssl.py(399)_generate_host_cert()
-> self._generate_ca()
 web-platform-tests/tools/wptserve/wptserve/sslutils/openssl.py(342)_generate_ca()
-> with self._config_openssl(None) as openssl:
  web-platform-tests/tools/wptserve/wptserve/sslutils/openssl.py(43)__enter__()
-> f.write(get_config(self.base_path, self.hosts, self.duration))
```

(interestingly `pdb` seems to be omitting `_config_openssl` in that!)

As such, this doesn't actually fix the underlying issue! I think we need to have a `is_ca` arg here?